pkg/apis/networking/validation/validation.go

```diff
@@ -68,11 +68,21 @@ func ValidateNetworkPolicyPort(port *networking.NetworkPolicyPort, portPath *fie
 			for _, msg := range validation.IsValidPortNum(int(port.Port.IntVal)) {
 				allErrs = append(allErrs, field.Invalid(portPath.Child("port"), port.Port.IntVal, msg))
 			}
+			if port.EndPort != nil && *port.EndPort < port.Port.IntVal {
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.IntVal, "endPort field must be equal or greater than port field"))
+			}
 		} else {
+			if port.EndPort != nil {
+				allErrs = append(allErrs, field.Invalid(portPath.Child("endPort"), port.Port.StrVal, "endPort cannot be used when port is non-numeric"))
```

Comment:
minor nit: to be consistent with other error messages , maybe something like "endPort cannot be defined when port is non-numeric"